### ISSUE TITLE
Automatically try domain fronting if direct connection to the Signal service fails

### ISSUE THREAD

**Issue by Rolf558** [WhisperSystems/Signal-Android#7279]
Hi there,

my father has traveled (from Gemany) to Egypt vor vacation. Although he has a working data connection, my messages over Signal are not delivered (single check marks only).

It looks like domain fronting does not turn on. Just my assumption...

(Btw. displaying connection state and in addition domain fronting on/off would be great.)

I am not sure what exact Signal version he is using. But I suppose one of the latest few (non-beta) releases.

**Comment by johanw666** [comment]
Domain fronting is currently only used when you register with a number from the country for which it is used.

So if your father needs it he should get a local sim and re-register Signal with that number. His identity key and message history will remain the same and after registration he can replace the local sim with his German one. When he is back he can re-register with his German sim.

**Comment by Rolf558** [comment]
@johanw666 thanks for the info. I think we move to an other messaging service in that case.

Binding domain fronting to the phone number makes IMHO no sense since plains have been invented. Why does Signal not automatically try domain fronting when it cannot connect to the service?

Let's use this issue for this request.

**Comment by moxie0** [comment]
Duplicate of an existing issue. In any case we're not going to do that, domain fronting is country specific, we don't know where you are, etc.

**Comment by dcherian** [comment]
@moxie0 You could ask the user to select a country from a popup list, after telling them that it looks like Signal has been blocked?

**Comment by moxie0** [comment]
There is no way to know that Signal has been censored. You could be on wifi at the mall without having clicked through the captive portal, there could be a signal outage, there could be a comcast outage, your linksys router might need to be restarted, etc.

**Comment by johanw666** [comment]
You can ask for the current mobile network and check the country code of the operator. If that is not available (wifi only, no simcard, permission denied, ...) a fallback to the number where Signal is registered with would be a logical option.

**Comment by dcherian** [comment]
> There is no way to know that Signal has been censored. You could be on wifi at the mall without having clicked through the captive portal, there could be a signal outage, there could be a comcast outage, your linksys router might need to be restarted, etc.

For everything but the Signal outage, you would be able to ping Google (if you can't get google, there's no connection / domain fronting wouldn't work anyway).

For the signal outage, using domain fronting should still throw an error leaving the user in the same position as before, i.e. unable to send anything.

**Comment by moxie0** [comment]
Imagine the shit storm of a 30 second Signal outage causing millions of people all over the world to get a notification that says "The country you are in may be censoring Signal. Please select your location." Not to mention that it would likely only make the outage worse if people did then try to connect through domain fronting. I'm sorry but we're not going to do this.

**Comment by Rolf558** [comment]
@moxie0 I understand. Thanks for your explanation.

**Comment by BastianBalthasarBux** [comment]
This issue popped up, when my wife went to Oman for holidays. Signal suddenly just is not usable anymore.
While I would overall blame badass countries with censorship for that, it still is also an issue for Signalapp itself.
IMHO it is just stupid, to bind domain-fronting feature to the phone number, since we have "portable" phones nowadays, you might have heard about, which fit nicely in cabin-luggage of modern planes (planes are objects which could fly and bring passengers and luggages in short time to the other side of the planet ...)!
(sorry for being sarcastic, but the excuses I read here, also here: https://github.com/signalapp/Signal-Android/issues/5986 , just make me - as a software developer - angry.)

**Comment by johanw666** [comment]
On the other hand, domain fronting seems to be blocked by all major cdn's at the moment because Telegram also used it and Google became blocked in Russia because of it. Besides that, the exact details of domain fronting differ per country so your requested functionality (which I certainly understand) would require code to check the country the network you are connected with is in. I'm not sure if this can be done reliably.

**Comment by BastianBalthasarBux** [comment]
In the end I start to regret my decision for signal ... no usable desktop client for linux anymore, that domain fronting thing etc ... and shame on me, i convinced a lot of people to use it ... maybe I should have chosen Threema ...

**Comment by BastianBalthasarBux** [comment]
> Besides that, the exact details of domain fronting differ per country so your requested functionality (which I certainly understand) would require code to check the country the network you are connected with is in. I'm not sure if this can be done reliably.

Add a Button in the extended settings menu: "I am currently in ...", and a dropdown to select one of the countries which is known to censor.
IMHO that's nothing which must be "automagically". It is ok to have some special configuration for it.
I could help by SMS to activate that feature for e.g. my wife.
The situation now is: I simply cannot help. Even if I would add that feature to the signal client on my own, it would be pretty complex to guide her through the process of installing an .apk from one of my servers. ...

**Comment by moxie0** [comment]
Hey @BastianBalthasarBux , we try not to use GH for feature requests or discussion. Please see the forums for that; you're more likely to see a positive outcome there.

Also, you should know that we don't allow verbal abuse, sarcasm, or negative and unhelpful communication here.
